rocksdb/columnfamily/cfdescriptor.nim

```diff
@@ -0,0 +1,44 @@
+# Nim-RocksDB
+# Copyright 2024 Status Research & Development GmbH
+# Licensed under either of
+#
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE) or http://www.apache.org/licenses/LICENSE-2.0)
+#  * GPL license, version 2.0, ([LICENSE-GPLv2](LICENSE-GPLv2) or https://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html)
+#
+# at your option. This file may not be copied, modified, or distributed except according to those terms.
+
+{.push raises: [].}
+
+import
+  ../internal/utils,
+  ./cfopts
+
+export cfopts
+
+type
+  ColFamilyDescriptor* = object
+    name: string
+    options: ColFamilyOptionsRef
+
+proc initColFamilyDescriptor*(
+    name: string,
+    options = defaultColFamilyOptions()): ColFamilyDescriptor =
+  ColFamilyDescriptor(name: name, options: options)
+
+template name*(descriptor: ColFamilyDescriptor): string =
```

Comment:
I'm using templates for performance reasons to reduce the overhead of invoking a proc but I'm not sure if my understanding is correct here. Will nim automatically optimize small procs and inline them anyway?